rocketpool/node/collectors/beacon-collector.go

```diff
@@ -178,7 +199,35 @@ func (collector *BeaconCollector) Collect(channel chan<- prometheus.Metric) {
 		collector.upcomingSyncCommittee, prometheus.GaugeValue, upcomingSyncCommittee)
 	channel <- prometheus.MustNewConstMetric(
 		collector.upcomingProposals, prometheus.GaugeValue, upcomingProposals)
+	channel <- prometheus.MustNewConstMetric(
+		collector.recentProposals, prometheus.GaugeValue, recentProposalCount)
+}
+
+func (collector *BeaconCollector) getProposedBlockCount(validatorIndices []string, head beacon.BeaconHead, slotsPerEpoch uint64) (float64, error) {
+	// prepare for quick lookups in event of many validators:
+	indexLookup := make(map[string]string, len(validatorIndices))
+	for _, index := range validatorIndices {
+		indexLookup[index] = index
+	}
+	latestSlot := head.FinalizedEpoch*slotsPerEpoch + (slotsPerEpoch - 1)
 
+	// check each block in the most recent epoch for our validators:
+	proposedBlockCount := float64(0)
+
+	for slot := latestSlot; slot > latestSlot-slotsPerEpoch; slot-- {
+		block, hasBlock, err := collector.bc.GetBeaconBlockHeader(strconv.FormatUint(slot, 10))
+		if err != nil {
+			collector.logError(fmt.Errorf("error getting beacon block: %w", err))
+			continue
+		}
+		if !hasBlock {
+			continue
+		}
+		if _, ok := indexLookup[block.ProposerIndex]; ok {
+			proposedBlockCount++
+		}
```

Comment:
```suggestion
		if _, ok := indexLookup[block.ProposerIndex]; !ok {
			continue
		}
		proposedBlockCount++
```

incredibly nitpicky, but i think:
```
for {
  if thing1 {
    continue
  }
  if thing2 {
    continue
  }
  if thing3 {
    continue
  }
  success()
}
```

is a more semantic expression here